src/sage/modular/hypergeometric_motive.py

```diff
@@ -1629,16 +1647,117 @@ def sign(self, t, p):
             sign = kronecker_symbol(t * (t - 1) * self._sign_param, p)
         return sign
 
+    def euler_factor_tame_contribution(self, t, p, mo, deg=None):
+        """
+        Return a contribution to the Euler factor of the motive `H_t` at a tame prime.
+
+        The output is only nontrivial when `t` has nonzero `p`-adic valuation.
+        The recipe is described in section 11.4.1 of [Watkins]_.
+
+        INPUT:
+
+        - ``t`` -- rational number, not 0 or 1
+
+        - ``p`` -- prime number of good reduction
+
+        - ``mo`` -- integer
+
+        - ``deg`` -- integer (optional)
+
+        OUTPUT:
+
+        a polynomial
+
+        If ``deg`` is specified, the output is truncated to that degree (inclusive).
```

Comment:
It would be good to have a doctest showing this. The `deg=1` of a degree 2 polynomial doesn't quite do this.

Reply:
This is now resolved by the doctest:
```
            sage: H.euler_factor_tame_contribution(11^20, 11, 5, deg=3)
            161051*T^3 + 6171*T^2 + 121*T + 1
```